### PR TITLE
Rewrite and synthetis the installation doc

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Add title for python 2 compatible version in changelog and add information
about installation for older interpreters. Refactor ide integration and put it in
the installation part for clarity.

Small reviewable part of #6589 and follow-up to #6613

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -54,6 +54,7 @@
     # "<source>": "<target>"
     "intro": "index.html",
     "support": "contact.html",
+    "ide-integration": "installation.html",
```
Comment (DanielNoord) [inline doc/conf.py:57]: I should have checked this earlier, but I don't think these actually work. The `RTD` output seems incorrect as well and the link:
https://pylint--6622.org.readthedocs.build/en/latest/user_guide/ide-integration.html
doesn't work.
https://pylint.pycqa.org/en/latest/user_guide/ide-integration.html

File: doc/whatsnew/index.rst [hunk 2]
```diff
@@ -24,6 +24,13 @@ High level descriptions of the most important changes between major Pylint versi
...
+
+Python 2 compatible version
+---------------------------
+
+.. toctree::
+   :maxdepth: 1
+
    1.9.rst
    1.8.rst
    1.7.rst
```
Comment (DanielNoord) [inline doc/whatsnew/index.rst:36]: Could you also add a Heading for `Changelog`? See:
https://pylint--6622.org.readthedocs.build/en/latest/whatsnew/index.html

File: doc/whatsnew/index.rst [hunk 3]
```diff
@@ -24,6 +24,13 @@ High level descriptions of the most important changes between major Pylint versi
    2.2.rst
    2.1.rst
    2.0.rst
+
+Python 2 compatible version
```
Comment (DanielNoord) [inline doc/whatsnew/index.rst:28]:
```suggestion
Python 2 compatible versions
```

File: doc/user_guide/ide_integration/visual-studio-code.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| refactor             | pylintCategorySeverity.refactor   | Hint             |
++----------------------+-----------------------------------+------------------+
+| warning              | pylintCategorySeverity.warning    | Warning          |
++----------------------+-----------------------------------+------------------+
+| error                | pylintCategorySeverity.error      | Error            |
++----------------------+-----------------------------------+------------------+
+| fatal                | pylintCategorySeverity.fatal      | Error            |
++----------------------+-----------------------------------+------------------+
+
+.. _Pylint command line arguments: https://pylint.pycqa.org/en/latest/user_guide/run.html#command-line-options
```
Comment (DanielNoord) [inline doc/user_guide/ide_integration/visual-studio-code.rst:57]: We should probably use an internal link here.

File: doc/user_guide/ide_integration/ide-integration.rst [hunk 5]
```diff
@@ -0,0 +1,12 @@
+###########################
+ Editor and IDE integration
+###########################
+
```
Comment (DanielNoord) [inline doc/user_guide/ide_integration/ide-integration.rst:4]:
```suggestion
Pylint can be integrated in various editors and IDE's. Below you can find tutorials for some of the most common ones.

```

File: doc/Makefile [hunk 6]
```diff
@@ -38,14 +38,12 @@ help:
...
-html:
-	@echo
-	@echo "Install dependencies"
-	$(PIP) install furo
-	@echo "Finished installing dependencies"
+build-html:
 	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html
+	@echo "HTML build complete"
+
+html: build-html
```
Comment (DanielNoord) [inline doc/Makefile:45]: Why split this?

File: doc/user_guide/installation.rst [hunk 7]
```diff
@@ -1,27 +1,75 @@
...
+
+.. code-block:: sh
+
+   pip install pylint
+
+
+Use the newer python interpreters if you can. It's possible to analyse code written
+for older interpreters by using ``py-version`` and setting it to the old interpreter. For
+example you can check that there are no ``f-string`` in python 3.5 code using python 3.8
+with an up-to-date pylint even if python 3.5 is past enf of life (EOL).
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:20]:
```suggestion
It is recommend to use the newest Python interpreter you can use. However, it is also possible to analyse code written
for older interpreters by using the ``py-version`` option and setting it to the old interpreter. For
example you can check that there are no ``f-strings`` in Python 3.5 code using Python 3.8
with an up-to-date pylint even if Python 3.5 is past end of life (EOL).
```

File: doc/user_guide/installation.rst [hunk 8]
```diff
@@ -1,27 +1,75 @@
...
 
-Python packages
-'''''''''''''''
+If a newer version does not work for you, we do not maintain pylint for EOL python,
+so the best available pylint might be an old version that works with your old interpreter but
+without the bug fixes and feature of latest pylints.
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:24]:
```suggestion
```

A bit redundant I think.

File: doc/user_guide/installation.rst [hunk 9]
```diff
@@ -1,27 +1,75 @@
...
 
-Pylint should be easily installable using pip.
+NB: You can also use ``conda`` or your system package manager on debian based OS.
+There package manager lag a little behind as it's maintained by a separate entity.
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:27]:
```suggestion
However, the available version can lag behind a little as it's maintained by a separate entity.
```

File: doc/user_guide/installation.rst [hunk 10]
```diff
@@ -1,27 +1,75 @@
...
-From the source distribution, extract the tarball, go to the extracted
-directory and simply run ::
+Editor integration
+------------------
 
-    python setup.py install
+.. _ide-integration:
 
-Or you can install it in editable mode, using ::
+- Eclipse_
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:43]:
```suggestion
For more explicit instructions for integration of pylint with editors and IDEs see: `ide_integration/ide-integration`_

- Eclipse_
```

Or however the internal link should be written.

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Change looks good itself! I like how stuff gets a more dedicated place.

File: doc/user_guide/ide_integration/visual-studio-code.rst (hunk 4 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/ide_integration/visual-studio-code.rst:57]: Nice catch, I just copy pasted / burst the previous content without checking.

File: doc/user_guide/installation.rst (hunk 7 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/installation.rst:20]: I think we should not try to advise users on the interpreter they should use for **their code** but stay in our lane an tell them what interpreter to use for pylint. (It's probably going to be the same than their code though let's be realist.)

File: doc/Makefile (hunk 6 above)
Comment (Pierre-Sassoulas) [reply doc/Makefile:45]: I opened  #6623 I wanted to remove the commit from this branch.

File: doc/conf.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply doc/conf.py:57]: ¨The previous redirect worked, I'll fix this one.

File: doc/user_guide/installation.rst [hunk 11]
```diff
@@ -1,27 +1,75 @@
...
+- Emacs_
+- `Eric IDE`_ in the `Project > Check` menu,
+- gedit_
+- `Jupyter Notebook`_
+- Komodo_
+- PyCharm_ :ref:`the section below <pylint_in_pycharm>`
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:49]:
```suggestion
- PyCharm_, see our:ref:`tutorial <pylint_in_pycharm>`
```

File: doc/user_guide/installation.rst [hunk 12]
```diff
@@ -1,27 +1,75 @@
...
+- PyDev_
+- pyscripter_ in the `Tool -> Tools` menu.
+- Spyder_ in the `View -> Panes -> Static code analysis`
+- TextMate_  :ref:`the section below <pylint_in_textmate>`
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:53]:
```suggestion
- TextMate_, see our  :ref:`tutorial <pylint_in_textmate>`
```

**Review by DanielNoord — CHANGES_REQUESTED** [review]

File: doc/user_guide/installation.rst (hunk 11 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/installation.rst]: I removed the link to the IDE, if you're searching a pylint plugin for it you don't need to learn about it.

File: doc/user_guide/installation.rst (hunk 12 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/installation.rst]: I removed the link to the IDE, if you're searching a pylint plugin for it you don't need to learn about it.

File: doc/user_guide/ide_integration/visual-studio-code.rst [hunk 13]
```diff
@@ -0,0 +1,62 @@
+.. _visual-studio-code:
+
+Integrate Pylint with Visual Studio Code
+========================================
+
+See external documentation:
+
+- Linting python: https://code.visualstudio.com/docs/python/linting
+- Pylint Extension: https://marketplace.visualstudio.com/items?itemName=ms-python.pylint
```
Comment (Pierre-Sassoulas) [inline doc/user_guide/ide_integration/visual-studio-code.rst:9]: Added the external doc here, I'm hesitating to simply remove our own doc and only link one of the link.

Comment (DanielNoord) [reply doc/user_guide/ide_integration/visual-studio-code.rst:9]: I'd be okay with removing it. There is a bit of duplication with other sections as well.

File: doc/user_guide/installation.rst [hunk 14]
```diff
@@ -1,27 +1,74 @@
...
+Use the newest Python interpreter if you can.
+
+It's possible to analyse code written for older interpreters by using the ``py-version``
+option and setting it to the old interpreter. For example you can check that there are
+no ``f-strings`` in Python 3.5 code using Python 3.8 with an up-to-date pylint even if
+Python 3.5 is past end of life (EOL).
 
-Python packages
-'''''''''''''''
+We do not guarantee that ``py-version`` will work for all EOL python interpreter indefinitely,
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:23]:
```suggestion
We do not guarantee that ``py-version`` will work for all EOL python interpreters indefinitely,
```

File: doc/user_guide/installation.rst [hunk 15]
```diff
@@ -1,27 +1,74 @@
...
+(for anything before python 3.5, it's probably not). If a newer version does not work for you,
+the best available pylint might be an old version that works with your old interpreter but
+without the bug fixes and feature of latest pylints.
```
Comment (DanielNoord) [inline doc/user_guide/installation.rst:26]:
```suggestion
without the bug fixes and feature of later versions.
```

File: doc/user_guide/ide_integration/visual-studio-code.rst (hunk 13 above)
Comment (Pierre-Sassoulas) [reply doc/user_guide/ide_integration/visual-studio-code.rst]: I removed the two that had better doc elsewhere.

**Pull request merged**